### PR TITLE
fix for transient failure in OffheapIncrementalIndex

### PR DESCRIPTION
**Pull request by himanshug**
before facts get it , indexAndOffsets should already know about it

fixes https://github.com/druid-io/druid/issues/2456

### CODE REVIEW

**Comment by fjy** [comment]
:+1: after travis

**Comment by himanshug** [comment]
I will send backport PR to 0.9 once merged.

File: processing/src/main/java/io/druid/segment/incremental/OffheapIncrementalIndex.java [hunk 1]
```diff
@@ -241,10 +245,11 @@ protected Integer addToFacts(
         }
 
         final Integer rowIndex = indexIncrement.getAndIncrement();
+        
+        indexAndOffsets.add(new int[]{bufferIndex, bufferOffset});
```
Comment (nishantmonu51) [inline processing/src/main/java/io/druid/segment/incremental/OffheapIncrementalIndex.java]: can you add a comment in the code explaining why this needs to be before adding to the facts?

**Comment by nishantmonu51** [comment]
:+1: after adding a small comment in code about ordering.

File: processing/src/main/java/io/druid/segment/incremental/OffheapIncrementalIndex.java (hunk 1 above)
Comment (himanshug) [inline processing/src/main/java/io/druid/segment/incremental/OffheapIncrementalIndex.java]: @nishantmonu51 added comment.

**Comment by fjy** [comment]
@himanshug needs backport

**Pull request merged**